src/NUnitFramework/framework/Internal/TestParameters.cs

```diff
@@ -123,23 +123,29 @@ private void InitializeArguments(object[] args)
 
         #endregion
 
-        #region IApplyToTest Members
-
         /// <summary>
-        /// Applies ParameterSet values to the test itself.
+        /// Applies the encapsulated parameters to the test method.
         /// </summary>
-        /// <param name="test">A test.</param>
-        public void ApplyToTest(Test test)
+        public void ApplyToTest(TestMethod test, TestNameGenerator defaultTestNameGenerator)
         {
-            if (this.RunState != RunState.Runnable)
-                test.RunState = this.RunState;
+            if (RunState != RunState.Runnable)
+                test.RunState = RunState;
 
             foreach (string key in Properties.Keys)
                 foreach (object value in Properties[key])
                     test.Properties.Add(key, value);
-        }
 
-        #endregion
+            if (TestName == null)
+            {
+                test.Name = defaultTestNameGenerator.GetDisplayName(test, OriginalArguments);
```

Comment:
I'll have to look elsewhere to understand this but in the original implementation, the test already has a name when it is created. The name is changed by `ApplyToTest` after the fact. In that way, the method doesn't need to have the default name generator passed to it.

If we do end up deciding to keep this, however, then let's change the name of the method so it doesn't cause confusion with the IApplyToTest method. Objects that implement IApplyToTest are supposed to only work with a small, local context and not have extra information available like whether the user specified a different default on the command line.

Reply:
We still need the default name generator because we pass the argument display names to it (seen in a later commit). It's either this, or make `ArgNames` public and keep the logic external- hopefully in a single place and not in each builder.

I'm happy with changing the method name. I almost did so.